doc/Manuals/LoadBalancing/ig-lb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,691 @@
+| health-check-port | `0` (disabled) | The tcp port the service listens to for HTTP requests. The default value `0` disables the service. |
+
+Below is a configuration that can be added to  `/etc/xroad/conf.d/local.ini` on the master that would enable the health check
+service on all the nodes once the configuration has been replicated. Changes to the settings require restarting the
+`xroad-proxy` service to take effect. This example enables listening to all available network interfaces (`0.0.0.0`) on
+port 5588.
+
+```
+[Proxy]
+health-check-interface=0.0.0.0
+health-check-port=5588
+```
+
+The service can be accessed using plain HTTP. It will return `HTTP 200 OK` if the proxy should be able to process messages
+and `HTTP 500 Internal Server Error` otherwise. A short message about the failure reason, if available, is added to the
+body of the response. The service runs as a part of the `xroad-proxy` service.
+
+In addition to implicitly verifying that the `xroad-proxy` service is running, the  health checks verify that:
+* The server authentication key is accessible and that the OCSP response for the certificate is `good`. This requires a
+running `xroad-signer` service in good condition.
+* The `serverconf` database is accessible.
+
+Each of these status checks has a separate timeout of 5 seconds. If the status check fails to produce a response in this
+time, it will be considered a health check failure and will cause a `HTTP 500` response.
+
+In addition, each status check result will be cached for a short while to avoid excess resource usage. A successful status
+check result will be cached for 2 seconds before a new verification is triggered. This is to make sure the OK results are
+as fresh as possible while avoiding per-request verification. In contrast, verification failures will be cached for 30
+seconds before a new verification is triggered. This should allow for the security server to get up and running after a
+failure or possible reboot before the status is queried again.
+
+
+#### 3.4.1 Known check result inconsistencies vs. actual state
+There is a known but rarely and not naturally occurring issue where the health check will report and OK condition for a
+limited time but sending some messages might not be possible. This happens when an admin user logs out of the keys.
+
+The health check will detect if the tokens (the key containers) have not been signed into after `xroad-signer` startup.
+It will however, not detect immediately when the tokens are manually logged out of. The keys are cached by the `xroad-proxy`
+process for a short while. As long as the authentication key is still cached, the health check will return OK. The necessary
+signing context values for sending a message might no longer be cached though. This means messages might fail to be sent
```

Comment:
> "As long as the authentication key is still cached, the health check will return OK. The necessary signing context values for sending a message might no longer be cached though."

  the word though should be used to connect the juxtaposing clauses "...will return OK, even though the..."